### ISSUE TITLE
Custom directive breaks make_executable_schema when Union is present

### ISSUE THREAD

**Issue by jimtheplant** [mirumee/ariadne#249]
Hi,

I'm fairly new to graphql/the ariadne library, but so far it is pretty awesome. However, I noticed an issue when trying to implement a custom directive in my project.

When i add a custom directive to a schema that has a Union in it, make_executable_schema seems to break. After some stack tracing I think it has to do with the heal_schema in schema_visitor.py.
When the type_ is a GraphQLUnionType, it's types field is a list. Since update_each_key is expecting a dictionary this causes some issues.

My code:
```
from ariadne import make_executable_schema, gql, QueryType, ObjectType, UnionType, SchemaDirectiveVisitor

schema_string = gql("""
union UnionTest = Type1 | Type2

type Query {
    hello: String
}

type Type1 {
    foo: String
}
type Type2 {
    bar: String
}

directive @customdirective on OBJECT | INTERFACE
""")

class CustomDirective(SchemaDirectiveVisitor):
    def visit_object(self, object_):
        pass
    def visit_interface(self, iface):
        pass

Type1 = ObjectType("Type1")
Type2 = ObjectType("Type2")

def resolve_union_test_type(obj, *_):
    if isinstance(obj, Type1):
        return "ValidationError"
    if isinstance(obj, Type2):
        return "AccessError"
    return None


query = QueryType()
UnionTest = UnionType("UnionTest", resolve_union_test_type)
make_executable_schema(schema_string, [query, Type1, Type2], directives={"customdirective": CustomDirective})
```
I've tested commenting out the update_each_key and it seems to work fine, but I'm unsure of how this would effect a directive that visits Unions

**Comment by rafalp** [comment]
Hello. Can you post the error you are experiencing here?

**Comment by jimtheplant** [comment]
I tried applying it to one of the types, however it still gave me the same issue. Get rid of the Union and it works fine.

Here's the stacktrace:
Traceback (most recent call last):
  File "C:/Users/Matt/PycharmProjects/federation-sandbox/bug.py", line 39, in <module>
    make_executable_schema(schema_string, [query, Type1, Type2], directives={"customdirective": CustomDirective})
  File "C:\Users\Matt\PycharmProjects\federation-sandbox\venv\lib\site-packages\ariadne\executable_schema.py", line 39, in make_executable_schema
    SchemaDirectiveVisitor.visit_schema_directives(schema, directives)
  File "C:\Users\Matt\PycharmProjects\federation-sandbox\venv\lib\site-packages\ariadne\schema_visitor.py", line 455, in visit_schema_directives
    heal_schema(schema)
  File "C:\Users\Matt\PycharmProjects\federation-sandbox\venv\lib\site-packages\ariadne\schema_visitor.py", line 585, in heal_schema
    heal(schema)
  File "C:\Users\Matt\PycharmProjects\federation-sandbox\venv\lib\site-packages\ariadne\schema_visitor.py", line 512, in heal
    each(original_type_map, _heal_type)
  File "C:\Users\Matt\PycharmProjects\federation-sandbox\venv\lib\site-packages\ariadne\schema_visitor.py", line 64, in each
    callback(value, key)
  File "C:\Users\Matt\PycharmProjects\federation-sandbox\venv\lib\site-packages\ariadne\schema_visitor.py", line 510, in _heal_type
    heal(named_type)
  File "C:\Users\Matt\PycharmProjects\federation-sandbox\venv\lib\site-packages\ariadne\schema_visitor.py", line 547, in heal
    update_each_key(type_.types, heal_type)
  File "C:\Users\Matt\PycharmProjects\federation-sandbox\venv\lib\site-packages\ariadne\schema_visitor.py", line 75, in update_each_key
    for key, value in object_map.items():
AttributeError: 'list' object has no attribute 'items'

Just a general question, but I thought directives can be defined but not used. Thanks for the quick response!

**Comment by 0x1EE7** [comment]
According to your stacktrace that particular issue was resolved in #245 probably distributed package was not updated. Can you please try to install from master? You are right directives can be defined but not used.
```python
def test_directive_can_be_defined_without_being_used():
    type_defs = """
        union UnionTest = Type1 | Type2

        type Query {
            hello: String
        }

        type Type1 {
            foo: String
        }
        type Type2 {
            bar: String
        }

        directive @customdirective on OBJECT | INTERFACE
    """

    class CustomDirective(SchemaDirectiveVisitor):
        def visit_object(self, object_):
            pass

        def visit_interface(self, iface):
            pass

    Type1 = ObjectType("Type1")
    Type2 = ObjectType("Type2")

    def resolve_union_test_type(obj, *_):
        if isinstance(obj, Type1):
            return "ValidationError"
        if isinstance(obj, Type2):
            return "AccessError"
        return None

    query = QueryType()
    UnionTest = UnionType("UnionTest", resolve_union_test_type)  # noqa
    make_executable_schema(
        type_defs,
        [query, Type1, Type2],
        directives={"customdirective": CustomDirective},
    )
```
```
.venv µ ❯❯❯ pytest tests/test_directives.py::test_directive_can_be_defined_without_being_used

==================================== test session starts ====================================
platform darwin -- Python 3.7.3, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
rootdir: /Users/null/dev/ariadne
plugins: django-3.5.1, asyncio-0.10.0, cov-2.7.1, mock-1.10.4, snapshottest-0.5.1
collected 1 item                                                                            

tests/test_directives.py .                                                            [100%]
================================= 1 passed in 0.07 seconds ==================================
```

**Comment by rafalp** [comment]
I've released latest master as [0.7.0a2](https://pypi.org/project/ariadne/0.7.0a2/). I'll try to find time today to wrap up the docs so we could release Ariadne 0.7.0 tomorrow or friday.
